server/http_hooks.go

```diff
@@ -72,8 +86,8 @@ func (p *Plugin) handleHello(w http.ResponseWriter, r *http.Request) {
 
 // The Authorization header should be an empty string if the request is by an
 // authenticated user.
-func (p *Plugin) handleCheckAuthHeader(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
-	isAuthenticatedUser := c.SessionId != ""
+func (p *Plugin) handleCheckAuthHeader(w http.ResponseWriter, r *http.Request) {
+	isAuthenticatedUser := r.Header.Get("Mattermost-User-ID") != ""
```

Comment:
Passing the `plugin.Context` to the handler became difficult, so I switched to using the `Mattermost-User-Id` in the request header instead of the context's `SessionId`.